hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java

```diff
@@ -326,6 +326,11 @@ public enum OperationStatusCode {
   public static final String COMPACTION_KV_MAX = "hbase.hstore.compaction.kv.max";
   public static final int COMPACTION_KV_MAX_DEFAULT = 10;
 
+  /** Parameter name for the scanner size limit to be used in compactions */
+  public static final String COMPACTION_SCANNER_SIZE_MAX =
+    "hbase.hstore.compaction.scanner.size.limit";
+  public static final long COMPACTION_SCANNER_SIZE_MAX_DEFAULT = 10 * 1024 * 1024L; // 10MB
```

Comment:
Thanks. I use 10MB for compaction because I think the region should be closed as soon as possible and avoid being blocked by compaction as much as possible, while there might some throttling strategies to limit the speed of compaction.

Reply:
Ok. I was concerned if it would have a performance implication, but I guess it should be fine. Any chance you've run this on a cluster? If so any change in how long a compaction takes or cpu used?

Reply:
NO. We haven't applied the pre [HBASE-27558](https://issues.apache.org/jira/browse/HBASE-27558) on our cluster, this one too. But the previous solution for this issue (using heartbeart cells return) has run on our cluster smoothly for about one year.